src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,120 @@
+    protected static final String DELIMITER_STORAGE = " :: ";
+    protected static final DateTimeFormatter DISPLAY_FORMAT = DateTimeFormatter.ofPattern("dd MMM YY HH:mm");
+    protected static final DateTimeFormatter READER_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+
+    private static final String COMPLETE = "\u2713";
+    private static final String NOT_COMPLETE = "\u2718";
+
+    private static final String STORE_COMPLETED = COMPLETE;
+    private static final String STORE_INCOMPLETE = NOT_COMPLETE;
+
+    protected String description;
+    protected boolean isComplete = false;
+
+    protected Task(String description) throws TaskException {
+        verifyArgs(description);
+        this.description = description;
+    }
+
+    protected Task(String description, boolean isComplete) throws TaskException {
+        verifyArgs(description);
+        this.description = description;
+        this.isComplete = isComplete;
+    }
+
+    /**
+     * Checks if the description is empty.
+     *
+     * @param description Description of the task.
+     * @throws TaskException If the description is empty.
+     */
+    private void verifyArgs(String description) throws TaskException {
+        if (description.equals("")) {
+            throw new TaskException("Did you provide any description for this todo task?");
+        }
+    }
+
+    /**
+     * Returns the type of the task.
+     */
+    public abstract TaskType getTaskType();
+
+    /**
+     * Indicates the task as completed.
+     *
+     * @return The calling instance.
+     */
+    public Task completeTask() {
+        isComplete = true;
+        return this;
+    }
+
+    /**
+     * Returns the completion for display purposes.
+     *
+     * @return "[✘]" or "[✓]"
+     */
+    protected String printCompletionFlag() {
+        String completionFlag = isComplete ? COMPLETE : NOT_COMPLETE;
+        return "[" + completionFlag + "]";
+    }
+
+    /**
+     * Returns the storage representation of the completion status.
+     *
+     * @return ✘ or ✓
+     */
+    protected String storeCompletionFlag() {
+        return isComplete ? STORE_COMPLETED : STORE_INCOMPLETE;
+    }
+
+    /**
+     * Converts the storage representation of the flag to completionStatus.
+     *
+     * @param flag Storage representation of a task.
+     * @return True if the given flag is ✓
+     */
+    protected static boolean decodeCompletionFlag(String flag) {
+        return flag.equals(STORE_COMPLETED);
+    }
+
+    /**
+     * Returns the storage representation.
+     *
+     * @return The storage representation.
+     */
+    public abstract String toStorageString();
+
+    /**
+     * Checks if the task's description contains the searchTerm.
+     *
+     * @param searchTerm Term to search with.
+     * @return True if the description has the given searchTerm.
+     */
+    public boolean descriptionContains(String searchTerm) {
+        return description.contains(searchTerm);
+    }
```

Comment:
Should the boolean method be renamed so as to follow the coding standard for boolean methods, for example, "hasSearchTerm", as the method is to check whether searchTerm is contained in a description?